src/message.c

```diff
@@ -2891,3 +2911,21 @@ int qd_message_stream_data_append(qd_message_t *message, qd_buffer_list_t *data)
     qd_compose_free(field);
     return rc;
 }
+
+
+void qd_message_set_Q2_unblocked_handler(qd_message_t *msg,
+                                         qd_message_Q2_unblocked_handler_t callback,
+                                         const qd_alloc_safe_ptr_t *ptr)
```

Comment:
Can we please please call the *ptr parameter as *sp or *safe_ptr like so - const qd_alloc_safe_ptr_t *safe_ptr OR const qd_alloc_safe_ptr_t *sp

Reply:
Yeah, I've refactored this a bit.   I'm not crazy about passing in an actual pointer as the context and having the callback also take a pointer.  This API would infer that the pointer passed in is handed back, which it isn't.

I want to avoid a pointer  here because that raises ownership issues, possibly cross threads.  I chose instead a pass by value API so it's clear that ownership of the safe pointer is not transferred but copied.